### PR TITLE
Feature/766 Create new Schedule B model

### PR DESCRIPTION
**Pull request by mjtravers**
#766

### CODE REVIEW

**Review by Elaine-Krauss-TCG — APPROVED** [review]

File: front-end/src/app/shared/services/transaction.service.ts
Comment (toddlees) [inline front-end/src/app/shared/services/transaction.service.ts:20]: the new app follows this pattern:
`/transactions/` for general endpoints
`/transactions/schedule-a/` for schedule a
`/transactions/schedule-b/` for schedule b

File: front-end/src/app/shared/models/transaction.model.ts
Comment (toddlees) [inline front-end/src/app/shared/models/transaction.model.ts]: i changed this to `string | undefined` for a couple reasons
- transaction_id is not a required field in the payload because of new transactions
- there was nothing gained by having this be null while the other fields were undefined
- by sending undefined, i was able to improve the transaction_id generation on the backend.

File: front-end/src/app/shared/models/transaction.model.ts
Comment (toddlees) [inline front-end/src/app/shared/models/transaction.model.ts:20]: `parent_transaction_id` had to change to `parent_transaction_object_id` for the ContentType workaround to work.  I'm seeing if there's a way to get around that

File: front-end/src/app/shared/models/scha-transaction.model.ts
Comment (toddlees) [inline front-end/src/app/shared/models/scha-transaction.model.ts:7]: If you moved `form_type` `filer_committee_id_number` `transaction_id` into the parent class Transaction, should we move these common fields as well? 
`back_reference_tran_id_number` `back_reference_sched_name` `entity_type` `election_code` `election_other_description` `conduit_name` `conduit_street1` `conduit_street2` `conduit_city` `conduit_state` `conduit_zip` `memo_code` `memo_text_description` `reference_to_si_or_sl_system_code_that_identifies_the_account`

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts
Comment (toddlees) [inline front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: what do you think of moving this aggregation_group assignment into `getPreviousTransaction` since it's derived from `transactionType` which you are now passing in.  We'd have to update service function to not make a network call if it's undefined, but maybe that's where the condition should be anyways right?  let the component ask the service "i'd like the previous transaction" and have the service decide, based on the args, how much work it should do to return a value.  It would make this observable flow easier to read for sure

**Review by toddlees — DISMISSED** [review]

File: front-end/src/app/shared/models/scha-transaction.model.ts
Comment (toddlees) [reply front-end/src/app/shared/models/scha-transaction.model.ts:7]: oh i bet you looked ahead at other schedules. nm

File: front-end/src/app/shared/services/transaction.service.ts
Comment (mjtravers) [reply front-end/src/app/shared/services/transaction.service.ts:20]: Ok, that works. Nice and clear.

File: front-end/src/app/shared/models/transaction.model.ts
Comment (mjtravers) [reply front-end/src/app/shared/models/transaction.model.ts]: This change has been made.

File: front-end/src/app/shared/models/transaction.model.ts
Comment (mjtravers) [reply front-end/src/app/shared/models/transaction.model.ts:20]: This change will have to wait until the API changes for sched B are merged into the develop branch. Will stay as is for this PR.

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts
Comment (mjtravers) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: Good suggestion. This refactoring has been made for this PR.

**Pull request merged**
